### PR TITLE
fix: Fix issues with use of async

### PR DESCRIPTION
**Pull request by cbornet**
Fix issues with use of async

### CODE REVIEW

File: src/backend/base/langflow/custom/custom_component/component.py [hunk 1]
```diff
@@ -506,11 +507,10 @@ def __call__(self, **kwargs):
...
         for key, _input in self._inputs.items():
-            if callable(_input.value):
-                result = _input.value()
-                if inspect.iscoroutine(result):
-                    result = await result
-                self._inputs[key].value = result
+            if asyncio.iscoroutinefunction(_input.value):
+                self._inputs[key].value = await _input.value()
+            elif callable(_input.value):
+                self._inputs[key].value = await asyncio.to_thread(_input.value())
```
Comment (cbornet) [inline src/backend/base/langflow/custom/custom_component/component.py:513]: It seems if the input value is a standard callable, there is a risk that it makes blocking operations.

File: src/backend/base/langflow/custom/custom_component/component.py [hunk 2]
```diff
@@ -718,10 +718,11 @@ async def _build_results(self):
                         _results[output.name] = output.value
                         result = output.value
                     else:
-                        result = method()
                         # If the method is asynchronous, we need to await it
                         if inspect.iscoroutinefunction(method):
-                            result = await result
+                            result = await method()
+                        else:
+                            result = await asyncio.to_thread(method)
```
Comment (cbornet) [inline src/backend/base/langflow/custom/custom_component/component.py:725]: output.method could block

File: src/backend/base/langflow/graph/vertex/base.py [hunk 3]
```diff
@@ -814,10 +814,7 @@ async def build(
             # Run steps
             for step in self.steps:
                 if step not in self.steps_ran:
-                    if inspect.iscoroutinefunction(step):
-                        await step(user_id=user_id, event_manager=event_manager, **kwargs)
-                    else:
-                        step(user_id=user_id, event_manager=event_manager, **kwargs)
+                    await step(user_id=user_id, event_manager=event_manager, **kwargs)
```
Comment (cbornet) [inline src/backend/base/langflow/graph/vertex/base.py:817]: the steps are currently only coroutines.

File: src/backend/base/langflow/services/chat/service.py [hunk 4]
```diff
@@ -63,5 +63,5 @@ async def clear_cache(self, key: str, lock: asyncio.Lock | None = None) -> None:
             lock (Optional[asyncio.Lock], optional): The lock to use for the cache operation. Defaults to None.
         """
         if isinstance(self.cache_service, AsyncBaseCacheService):
-            return await self.cache_service.get(key, lock=lock or self.async_cache_locks[key])
+            return await self.cache_service.delete(key, lock=lock or self.async_cache_locks[key])
```
Comment (cbornet) [inline src/backend/base/langflow/services/chat/service.py:66]: Oups! This can't be good!

File: src/backend/base/langflow/services/manager.py [hunk 5]
```diff
@@ -95,9 +94,7 @@ async def teardown(self) -> None:
                 continue
             logger.debug(f"Teardown service {service.name}")
             try:
-                result = service.teardown()
-                if asyncio.iscoroutine(result):
-                    await result
+                await service.teardown()
```
Comment (cbornet) [inline src/backend/base/langflow/services/manager.py:97]: service.teardown is a coroutine

File: src/backend/base/langflow/custom/custom_component/component.py (hunk 1 above)
Comment (ogabrielluiz) [reply src/backend/base/langflow/custom/custom_component/component.py:513]: Should we be calling _input.value inside `to_thread` or passing the callable?
```suggestion
                self._inputs[key].value = await asyncio.to_thread(_input.value)
```

File: src/backend/base/langflow/services/chat/service.py (hunk 4 above)
Comment (ogabrielluiz) [reply src/backend/base/langflow/services/chat/service.py:66]: good catch!

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]
A part from the comment, LGTM

File: src/backend/base/langflow/custom/custom_component/component.py (hunk 1 above)
Comment (cbornet) [reply src/backend/base/langflow/custom/custom_component/component.py]: Good catch, thanks!

**Review by ogabrielluiz — APPROVED** [review]
LGTM

**Pull request merged**